auth/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Controller,
+  Body,
+  Post,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  SerializeOptions,
+  HttpCode,
+  HttpStatus,
+  Get,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { plainToInstance } from 'class-transformer';
+import { UserEntity } from '../db/entities/users.entity';
+import { AuthService } from './auth.service';
+import { LoginDTO } from './dtos/login.dto';
+import { SignUpDTO } from './dtos/signup.dto';
+import { RefreshTokenGuard } from './guards/refresh-token.guard';
+import { RequestWithRefreshPayload } from './interfaces/request-with-refresh-payload.interface';
+import { TokenResponse } from './responses/token.response';
+
```

Comment:
Wanna use Swagger or nah?
I mean REST is probably fine too, but then the frontend doesn't see response types like in swagger.

Reply:
I would add Swagger in a following issue. As of now the Auth API is not accessible for frontend yet and any changes in the API would also require changing the mess that is Swagger.

So yes I will add it later but it is out of scope here